src/script/miniscript.h

```diff
@@ -1188,23 +1188,28 @@ struct Node {
     //! Return the number of ops in the script (not counting the dynamic ones that depend on execution).
     uint32_t GetStaticOps() const { return ops.count; }
 
-    //! Check the ops limit of this script against the consensus limit.
-    bool CheckOpsLimit() const { return GetOps() <= MAX_OPS_PER_SCRIPT; }
+    //! Check the ops limit of this script against the consensus limit. This limit is only present for P2WSH.
+    template <typename Ctx> bool CheckOpsLimit(const Ctx& ctx) const {
+        return ctx.MsContext() == MiniscriptContext::TAPSCRIPT || GetOps() <= MAX_OPS_PER_SCRIPT;
+    }
```

Comment:
It might be worth explicitly enumerating both the known contexts, and have a defensive final  `assert(false);`.
Similarly for other places where the context is accessed).

Reply:
I could have a `IsTapscript(ctx)` helper that does that next to the `MiniscriptContext` enum. It could then be used in all those places to not have to duplicate the verbose `switch () ... assert(false)`.

Reply:
Alright, finally applied the change to the many occurrences in each commit. It should also be more readable now.